src/util/DeterministicFaultyGraphBuilder.java

```diff
@@ -0,0 +1,84 @@
+package util;
+
+import org.apache.commons.math3.util.CombinatoricsUtils;
+
+import java.util.*;
+
+public class DeterministicFaultyGraphBuilder {
+
+    static public Graph generateGraph(int rows, int columns, int index) {
+        Graph goldenGraph = RegularGraphBuilder.generateGraph(rows, columns);
+
+        if (index == 0)
+            return goldenGraph;
+
+        if (index > numberOfTopology(rows, columns))
+            throw new IndexOutOfBoundsException("Index " + index + " is out of bounds!");
+
+        List<Edge> listEdgesOfGraph = goldenGraph.getEdges();
+
+        for (int i = 0; i < listEdgesOfGraph.size(); i++) {
+            Vertex source = listEdgesOfGraph.get(i).source();
+            Vertex destination = listEdgesOfGraph.get(i).destination();
+
+            for (int j = i + 1; j < listEdgesOfGraph.size(); j++) {
+                if ((listEdgesOfGraph.get(j).source().equals(destination))
+                        && (listEdgesOfGraph.get(j).destination().equals(source))) {
+                    listEdgesOfGraph.remove(j);
+                }
+            }
+        }
+
+        Set<Edge> setGoldenGraph = new HashSet<Edge>(listEdgesOfGraph);
+        List<Set> listGraphsCombinations = Combinatorics.allCombinationsOf(setGoldenGraph);
+
+        List sortedListGraphsCombinations = new ArrayList(listGraphsCombinations);
+        Collections.sort(sortedListGraphsCombinations, new Comparator<Set>() {
+            public int compare(Set o1, Set o2) {
+                return o1.size() - o2.size();
+            }
+        });
+
+        Set<Edge> setGoldenGraphIndex = (Set<Edge>) sortedListGraphsCombinations.get(index);
+
+        for (Edge s : setGoldenGraphIndex) {
+            Edge t = goldenGraph.adjunct(s.destination(), s.source());
+            goldenGraph.removeEdge(s);
+            goldenGraph.removeEdge(t);
+        }
+
+        return goldenGraph;
+    }
+
+    private static int numberOfEdges(int rows, int columns) {
+        return (columns - 1) * rows + (rows - 1) * columns;
+    }
+
+    private static int numberOfMinGoodEdges(int rows, int columns) {
+        return (rows * columns) - 1;
+    }
+
+    private static int maxFaultyOfGraph(int rows, int columns) {
+        return numberOfEdges(rows, columns) - numberOfMinGoodEdges(rows, columns);
+    }
+
+    private static int numberOfGraphsForNumberFaulty(int rows, int columns, int nmFaulty) {
+        int n = numberOfEdges(rows, columns);
+        int k = nmFaulty;
+        Integer result = (int) CombinatoricsUtils.binomialCoefficient(n, k);
```

Comment:
That dependency should be hidden at 'util.Combinatorics'. Better than that we could implement it.